xbmc/windowing/egl/EGLNativeTypeIMX.cpp

```diff
@@ -273,17 +292,25 @@ bool CEGLNativeTypeIMX::ProbeResolutions(std::vector<RESOLUTION_INFO> &resolutio
 
   std::string valstr;
   SysfsUtils::GetString("/sys/class/graphics/fb0/modes", valstr);
-  std::vector<std::string> probe_str;
-  probe_str = StringUtils::Split(valstr, "\n");
+  std::vector<std::string> probe_str = StringUtils::Split(valstr, "\n");
+
+  // lexical order puts the modes list into our preferred
+  // order and by later filtering through FindMatchingResolution()
+  // we make sure we read _all_ S modes, following U and V modes
+  // while list will hold unique resolutions only
+  std::sort(probe_str.begin(), probe_str.end());
 
   resolutions.clear();
   RESOLUTION_INFO res;
   for (size_t i = 0; i < probe_str.size(); i++)
   {
-    if(!StringUtils::StartsWith(probe_str[i], "S:"))
+    if(!StringUtils::StartsWith(probe_str[i], "S:") && !StringUtils::StartsWith(probe_str[i], "U:") &&
+       !StringUtils::StartsWith(probe_str[i], "V:"))
       continue;
+
     if(ModeToResolution(probe_str[i], &res))
-      resolutions.push_back(res);
+      if(!FindMatchingResolution(res, resolutions))
```

Comment:
@koying 

in regards your resolution question:
we do that "duplicity strip off" primarily to workaround again mxc_hdmi issue. at the same time from perspective of the question if we unintentionally mask relevant modes (different by different clocking) we can say practically no - due to rules and workflow of mxc_hdmi driver in regards to S,U and V modes.

- S modes are provided in HDMI mode and do come from internal CEA table (& of course EDID signalling support of modes with specific VIC). also the CEA resolutions are supported only up to #47 with a many missing. when mxc_hdmi runs in HDMI mode, it won't pass any U or V modes into fb allowed resolutions

- in DVI mode, it won't pass any CEA(S) modes (as EDID extended CEA block wasn't read - and that's why we run in DVI mode). the U and V modes are parsed from primary EDID block by FB core code where U modes are again delivered from internal table by matching EDID records pointing it's support. V modes could be specified with full details, but FB driver gets xres, total clock and recalculates rest on it's own according to VESA GTF rules.

so all that combined puts as on the safe side as the specific mode tables either not overlap, or mxc_hdmi doesn't combine such tables which could overlap.
there also needs to be said that mxc_hdmi doesn't allow any other resolutions be set as the listed in /sys/..../modes and also doesn't read/allow resolutions not being in EDID and/or is not present in its mode tables.

related question to grab more insights - if we would parse resolutions different on timings, equal on the base WxHxR data. how KODI would process that ? all it's internal functions doing picking / comparing / enumerating the list are not going that deep to be able to distinguish at level of timings.
or am I mistaken?